pkg/operator/status.go

```diff
@@ -289,9 +293,82 @@ func (optr *Operator) syncUpgradeableStatus() error {
 		coStatus.Message = "One or more machine config pools are updating, please see `oc get mcp` for further details"
 	}
 
+	// don't overwrite status if we didn't run
+	if ran, err := optr.isKubeletSkewTooFar(); ran && err != nil {
+		if coStatus.Status != configv1.ConditionFalse {
+			coStatus.Reason = "KubeletSkewTooFar"
+			coStatus.Message = err.Error()
+		} else if err != nil {
+			coStatus.Reason = coStatus.Reason + "::" + "KubeletSkewTooFar"
+			coStatus.Message = coStatus.Message + "\n" + err.Error()
+		}
+	}
+
 	return optr.updateStatus(co, coStatus)
 }
 
+func (optr *Operator) isKubeletSkewTooFar() (bool, error) {
+	nodes, err := optr.nodeLister.List(labels.Everything())
+	if err != nil {
+		return true, err
+	}
+	kubeAPIServerStatus, err := optr.configClient.ConfigV1().ClusterOperators().Get(context.TODO(), "kube-apiserver", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return false, err
+	}
+	if err != nil {
+		return true, err
+	}
+
+	// looks like
+	//   - name: kube-apiserver
+	//     version: 1.21.0-rc.0
+	kubeAPIServerVersion := ""
+	for _, version := range kubeAPIServerStatus.Status.Versions {
+		if version.Name != "kube-apiserver" {
+			continue
+		}
+		kubeAPIServerVersion = version.Version
+		break
+	}
+	if len(kubeAPIServerVersion) == 0 {
+		return false, fmt.Errorf("kube-apiserver does not yet have a version")
+	}
+	kubeAPIServerMinorVersion, err := getMinorVersion(kubeAPIServerVersion)
+	if err != nil {
+		return true, err
+	}
+
+	skewedNodeErrors := []error{}
+	for _, node := range nodes {
+		// looks like kubeletVersion: v1.21.0-rc.0+6143dea
+		kubeletVersion := node.Status.NodeInfo.KubeletVersion
+		if len(kubeletVersion) == 0 {
+			continue
+		}
+		nodeMinorVersion, err := getMinorVersion(kubeletVersion)
+		if err != nil {
+			skewedNodeErrors = append(skewedNodeErrors, fmt.Errorf("node/%v has malformed kubelet version %q: %v", node.Name, kubeletVersion, err))
```

Comment:
what does this mean to a user? what are they supposed to do to fix?

Reply:
This only provides the user with more information that there's an error when checking the version. Might be helpful debugging if they run into other issues but no way to fix it at this point. Should we `glog.Error(err)` this error or silently ignore it?

Reply:
Would this error manifest as another error that already exists? IE how does a user get into this situation and would it already be reflected in the state of the pool (would the pool be degraded, etc..)

Reply:
> Would this error manifest as another error that already exists? IE how does a user get into this situation and would it already be reflected in the state of the pool (would the pool be degraded, etc..),would it fail the kubelet health check, etc..

I agree with the above situation. I tend to only report the message and reason if the version has been successfully checked and compared to verify there's a version skew.